broker/src/main/java/io/camunda/zeebe/broker/Broker.java

```diff
@@ -87,30 +92,32 @@
 import java.util.List;
 import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.TimeUnit;
+import java.util.function.BiFunction;
 import java.util.stream.Collectors;
 import org.slf4j.Logger;
 
 public final class Broker implements AutoCloseable {
 
   public static final Logger LOG = Loggers.SYSTEM_LOGGER;
-  private static final List<PartitionStep> LEADER_STEPS =
+
+  private static final List<PartitionBootstrapStep> PARTITION_BOOTSTRAP_STEPS =
       List.of(
-          new SnapshotReplicationPartitionStep(),
-          new StateControllerPartitionStep(),
-          new LeaderPostStoragePartitionStep(),
-          new LogDeletionPartitionStep(),
-          new LogStreamPartitionStep(),
-          new ZeebeDbPartitionStep(),
-          new StreamProcessorPartitionStep(),
-          new SnapshotDirectorPartitionStep(),
-          new RocksDbMetricExporterPartitionStep(),
-          new ExporterDirectorPartitionStep());
-  private static final List<PartitionStep> FOLLOWER_STEPS =
+          new SnapshotReplicationPartitionBootstrapStep(),
+          new StateControllerPartitionBootstrapStep(),
+          new LogDeletionPartitionBootstrapStep(),
+          new LogStreamPartitionBootstrapStep(),
```

Comment:
General I like the idea, but this will not work for multiple of these steps e.g. LogStream, needs to be reopened.